File: java/snowworld/solve.java
import java.util.*;


class Edge
{
    int to;
    int from;
    double snow;    
    public Edge(int to, int from, double snow){
        this.to = to;
        this.from = from;
        this.snow = snow;
    }    
    public int getTo() {
        return to;
    }
    public int getFrom() {
        return from;
    }
    public double getSnow() {
        return snow;
    }
}


class solve
{
    static ArrayList<Edge> getEdgeList(int e, Scanner reader){
        //Scanner reader = new Scanner(System.in);    
        ArrayList<Edge> edges = new ArrayList<Edge>();
        for(int i = 0; i < e; i++){
            edges.add(new Edge(reader.nextInt(), reader.nextInt(), 
                        reader.nextDouble()));  
        }
        return edges;
        
    }
    static int[][] adjacencyMatrix(int n, ArrayList<Edge> edges){
        int adjMat[][] = new int[n][n];
        for(int i = 0; i < edges.size(); i++){
            adjMat[edges.get(i).getTo()][edges.get(i).getFrom()] = 1;
            adjMat[edges.get(i).getFrom()][edges.get(i).getTo()] = 1;
        }
        return adjMat;
    }  
    static void solve(){
        /*Unimplemented */
        return;
    }
    
    
    public static void main (String[] args)
    {
        Scanner reader = new Scanner(System.in);
        int n;
        int e;
        double alpha;
        n = reader.nextInt();
        e = reader.nextInt();
        alpha = reader.nextDouble();
        
        ArrayList<Edge> edges = getEdgeList(e, reader);
        int[][] adjMat = adjacencyMatrix(n, edges);        
        
        solve();

        return;
    }
    
    
    
}

File: cpp/snowworld/solve.cpp
#include <cstdio>
#include <cstdlib>
#include <iostream>
#include <vector>
#include "math.h"

using namespace std;


typedef vector<pair<pair<int, int>, double> > edgeList;
edgeList getEdgeList(int e); 
int **adjacencyMatrix(int n, edgeList edges); 
void freeMatrix(int **adjMat, int n); 
void solve();                        

int main(){
  int n, e;
  double alpha;
  scanf("%d %d %lf", &n, &e, &alpha);
  edgeList edges = getEdgeList(e);
  int** adjMat = adjacencyMatrix(n, edges);
  
  solve();

  freeMatrix(adjMat, n);
  return 0;
}

/* Solves and prints out solution */

void solve(){
  /* Unimplemented */
}

/* Given functions */

/* Reads from stdin and forms an edgeList of type edgeList */
edgeList getEdgeList(int e){
  int a, b;
  double c;
  edgeList edges;
  for(int i = 0; i < e; i++){
    scanf("%d %d %lf", &a, &b, &c);
    edges.push_back(make_pair(make_pair(a,b), c));
  }
  return edges;
}

/* Creates an adjacency matrix from the edgeList */
int **adjacencyMatrix(int n, edgeList edges){
  int **adjMat = new int*[n];
  for(int i = 0; i < n; i++){
    adjMat[i] = new int[n];
  }
  int to, from;
  for(int i = 0; i < edges.size(); i++){
    to = edges[i].first.first;
    from = edges[i].first.second;
    adjMat[to][from] = 1;
    adjMat[from][to] = 1;
  }
  return adjMat;
}

/* Deletes what was allocated (using delete) */
void freeMatrix(int **adjMat, int n){
  if(adjMat == NULL) return;
  for(int i = 0; i < n; i++){
    delete [] adjMat[i];
  }
  delete [] adjMat;
}
